src/solver/conv_hip_implicit_gemm_bwd_v1r1_xdlops_nchw_kcyx_nkhw.cpp

```diff
@@ -0,0 +1,978 @@
+#endif
+
+        // this is the grid size using the biggest blockwise-GEMM
+        auto grid_size_max_blockwise_gemm =
+            (std::size_t(gemm_m) * gemm_n) / max_blockwise_gemm_size;
+
+        const float ratio = float(grid_size) / grid_size_max_blockwise_gemm;
+
+        if(grid_size_max_blockwise_gemm > 600)
+        {
+            if(ratio > 1.41)
+                return false;
+        }
+        if(grid_size_max_blockwise_gemm > 480)
+        {
+            if(ratio > 1.81)
+                return false;
+        }
+        if(grid_size_max_blockwise_gemm > 360)
+        {
+            if(ratio > 2.21)
+                return false;
+        }
+        if(grid_size_max_blockwise_gemm > 240)
+        {
+            if(ratio > 3.21)
+                return false;
+        }
+        else if(grid_size_max_blockwise_gemm > 120)
+        {
+            if(ratio > 6.21)
+                return false;
+        }
+    }
+
+    // don't need too many waves per block
+    {
+        const int wave_per_block = (GemmMPerBlock / GemmMPerWave) * (GemmNPerBlock / GemmNPerWave);
+
+        if(!(wave_per_block > 1 && wave_per_block <= 4))
+        {
+            return false;
+        }
+    }
+
+    // avoid skinny blockwise GEMM whenever possible
+    {
+        int gemm_m = 0;
+        int gemm_n = 0;
+
+        std::tie(std::ignore, gemm_m, gemm_n, std::ignore) =
+            ConvHipImplicitGemmForwardV4R4Xdlops::CalculateGemmSize(ctx);
+
+        if(GemmMPerBlock > 2 * GemmNPerBlock)
+        {
+            if(gemm_n % (2 * GemmNPerBlock) == 0)
+                return false;
+        }
+
+        if(GemmNPerBlock > 2 * GemmMPerBlock)
+        {
+            if(gemm_m % (2 * GemmMPerBlock) == 0)
+                return false;
+        }
+    }
+
+    // avoid skinny wavewise GEMM whenever possible
+    {
+        if(GemmMPerWave > 2 * GemmNPerWave)
+        {
+            if(GemmNPerBlock % (2 * GemmNPerWave) == 0)
+                return false;
+        }
+
+        if(GemmNPerWave > 2 * GemmMPerWave)
+        {
+            if(GemmMPerBlock % (2 * GemmMPerWave) == 0)
+                return false;
+        }
+    }
+
+    // each thread should not too much data
+    {
+        const int block_size = (GemmMPerBlock / GemmMPerWave) * (GemmNPerBlock / GemmNPerWave) * 64;
+
+        const int a_data_per_thread_copy = (GemmKPerBlock * GemmMPerBlock * GemmKPack) / block_size;
+        const int b_data_per_thread_copy = (GemmKPerBlock * GemmNPerBlock * GemmKPack) / block_size;
+
+        if(ctx.IsFp32())
+        {
+            if(a_data_per_thread_copy > 16 || b_data_per_thread_copy > 16)
+                return false;
+        }
+        else if(ctx.IsFp16() || ctx.IsBfp16())
+        {
+            if(a_data_per_thread_copy > 32 || b_data_per_thread_copy > 32)
+                return false;
+        }
+    }
+
+    // GemmKPerBlock*GemmKPack should not be too small, otherwise read performance of A matrix would
```

Comment:
This is true for fwd-v4r4, because ```GemmK``` and ```GemmKPack``` come from ```C*Y*X```.

But this is no true for bwd-data-v1r1, because ```GemmK``` and ```GemmKPack``` come from ```K```.

Reply:
Will do